### PR TITLE
lib/syscall_shim: Add `syscall` alias to `uk_syscall`

### PR DESCRIPTION
**Pull request by jschlumpp**
### Prerequisite checklist

 - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
 - [x] Tested your changes against relevant architectures and platforms;
 - [x] Ran the [`checkpatch.pl`](https://github.com/unikraft/unikraft/blob/staging/support/scripts/checkpatch.pl) on your commit series before opening this PR;
 - [x] Updated relevant documentation.

### Base target

 - Architecture(s): N/A
 - Platform(s): N/A
 - Application(s): N/A

### Additional configuration

 - `CONFIG_LIBSYSCALL_SHIM=y`

### Description of changes

The symbol is required for binaries that were compiled against a Linux musl and use musl's system call wrapper.

### CODE REVIEW

**Review by StefanJum — COMMENTED** [review]
I tested it and it seems to work fine, I'll test it using the elf loader sometime in the next few days.

File: lib/syscall_shim/uk_syscall.awk [hunk 1]
```diff
@@ -56,4 +56,5 @@ END {
 	printf "\tva_end(ap);\n"
 	printf "\treturn ret;\n"
 	printf "}\n"
+	printf "long syscall() __attribute__ ((weak, alias (\"uk_syscall\")));\n"
```
Comment (razvanvirtan) [inline lib/syscall_shim/uk_syscall.awk:59]: Is it an extra `)` at the end of the line?

Comment (jschlumpp) [reply lib/syscall_shim/uk_syscall.awk:59]: No?

Comment (razvanvirtan) [reply lib/syscall_shim/uk_syscall.awk:59]: My bad, sorry for this.

**Review by StefanJum — APPROVED** [review]
All good.
Reviewed-by: Stefan Jumarea <stefanjumarea02@gmail.com>

**Review by razvanvirtan — APPROVED** [review]
All good, thank you for this!

Reviewed-by: Razvan Virtan <virtanrazvan@gmail.com>

**Review by razvand — APPROVED** [review]
Approved-by: Razvan Deaconescu <razvand@unikraft.io>

**Pull request closed without merging**